File: smash-ranks-android/app/src/main/java/com/garpr/android/views/RegionSelectionItemView.java
package com.garpr.android.views;

import android.annotation.TargetApi;
import android.app.Activity;
import android.content.Context;
import android.os.Build;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.TextView;

import com.garpr.android.R;
import com.garpr.android.adapters.BaseAdapterView;
import com.garpr.android.misc.MiscUtils;
import com.garpr.android.models.Region;

import butterknife.BindView;
import butterknife.ButterKnife;

public class RegionSelectionItemView extends LinearLayout implements BaseAdapterView<Region>,
        View.OnClickListener {

    private Region mContent;

    @BindView(R.id.radioButton)
    RadioButton mRadioButton;

    @BindView(R.id.tvId)
    TextView mId;

    @BindView(R.id.tvDisplayName)
    TextView mDisplayName;


    public RegionSelectionItemView(final Context context, @Nullable final AttributeSet attrs) {
        super(context, attrs);
    }

    public RegionSelectionItemView(final Context context, @Nullable final AttributeSet attrs,
            final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public RegionSelectionItemView(final Context context, final AttributeSet attrs,
            final int defStyleAttr, final int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public Region getContent() {
        return mContent;
    }

    @Override
    public void onClick(final View v) {
        final Activity activity = MiscUtils.optActivity(getContext());

        if (activity instanceof Listeners) {
            ((Listeners) activity).onClick(this);
        }
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        ButterKnife.bind(this);
        setOnClickListener(this);
    }

    @Override
    public void setContent(final Region content) {
        mContent = content;

        mDisplayName.setText(mContent.getDisplayName());
        mId.setText(mContent.getId());

        final Activity activity = MiscUtils.optActivity(getContext());
        if (activity instanceof Listeners) {
            mRadioButton.setChecked(mContent.getId().equalsIgnoreCase(
                    ((Listeners) activity).getSelectedRegion()));
        } else {
            mRadioButton.setChecked(false);
        }
    }


    public interface Listeners {
        @Nullable
        String getSelectedRegion();

        void onClick(final RegionSelectionItemView v);
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/views/RegionSelectionItemView.kt
package com.garpr.android.views

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.constraintlayout.widget.ConstraintLayout
import com.garpr.android.features.base.BaseAdapterView
import com.garpr.android.data.models.Region
import com.garpr.android.extensions.activity
import com.garpr.android.extensions.clear
import kotlinx.android.synthetic.main.item_region_selection.view.*

class RegionSelectionItemView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null
) : ConstraintLayout(context, attrs), BaseAdapterView<Region>, View.OnClickListener {

    interface Listeners {
        fun onClick(v: RegionSelectionItemView)
        val selectedRegion: Region?
    }

    private fun clear() {
        radioButton.isChecked = false
        displayName.clear()
        regionId.clear()
    }

    override fun onClick(v: View) {
        (activity as? Listeners?)?.onClick(this)
    }

    override fun onFinishInflate() {
        super.onFinishInflate()
        setOnClickListener(this)
    }

    var region: Region? = null
        private set(value) {
            field = value

            if (value == null) {
                clear()
                return
            }

            radioButton.isChecked = value == (activity as? Listeners?)?.selectedRegion
            displayName.text = value.displayName
            regionId.text = value.id
        }

    override fun setContent(content: Region) {
        region = content
    }

}
